src/iexamine.cpp

```diff
@@ -5752,8 +5743,106 @@ static void mill_activate( Character &you, const tripoint &examp )
     map_stack items = here.i_at( examp );
     units::volume food_volume = 0_ml;
 
+    std::map<itype_id, int> millable_counts;
+
+    for( const item &iter : items ) {
+        if( iter.type->milling_data && !iter.type->milling_data->into_.is_null() ) {
+            if( millable_counts.find( iter.typeId() ) == millable_counts.end() ) {
+                millable_counts.emplace( iter.typeId(), 1 );
+            } else {
+                millable_counts[iter.typeId()]++;
+            }
+        }
+    }
+
+    for( std::pair<const string_id<itype>, int> mill_type_count : millable_counts ) {
+        item source( mill_type_count.first );
+        const item product( source.type->milling_data->into_ );
+        recipe rec;
+
+        for( std::map<recipe_id, recipe>::const_iterator iter = recipe_dict.begin();
+             iter != recipe_dict.end(); iter++ ) {
+            if( iter->first == source.type->milling_data->recipe_ ) {
+                rec = iter->second;
+                break;
+            }
+        }
```

Comment:
You don't have to do this lookup manually. A `recipe_id` already works like a pointer to a recipe, so simply dereferencing it will do that for you.